File: Java-Hospital-SOAP/src/main/java/ru/hazov/javahospitalsoap/entity/Disease.java
package ru.hazov.javahospitalsoap.entity;

import jakarta.persistence.*;
import jdk.jfr.Description;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Description("Болезнь")
public class Disease {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String name;

    @Column
    private String description;


    @OneToMany(mappedBy = "disease", fetch = FetchType.LAZY)
    List<PatientDisease> patients;

}

File: CSharpHospitalREST/CSharpHospitalREST/Models/Disease.cs
namespace CSharpHospitalREST.Models
{
    public class Disease
    {
        public long Id { get; set; }
        public string Name { get; set; }

        public string? Description { get; set; }

        public List<PatientDisease> PatientDiseases { get; set; } = new List<PatientDisease>();
    }
}
